gui/src/components/AutoScrollContainer.tsx

```diff
@@ -0,0 +1,275 @@
+import { forwardRef, useCallback, useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import { ChevronDownIcon } from "@heroicons/react/24/outline"; // Add this import
+import { lightGray, vscBackground } from ".";
+
+const ScrollContainer = styled.div<{ inputBoxHeight: number }>`
+  height: ${(props) => `calc(100% - ${props.inputBoxHeight}px)`};
+  display: flex;
+  flex-direction: column;
+  overflow-x: hidden;
+  overflow-y: hidden;
+  position: relative;
+  width: 100%;
+  margin-top: 0;
+  padding-top: 0;
+`;
+
+export const ScrollContent = styled.div`
+  contain: content;
+  flex: 0 1 auto;
+  overflow-y: auto;
+  overflow-x: hidden;
+  min-height: 0;
+  z-index: 98;
+  width: 100%;
+  margin-top: 0;
+  padding-top: 0;
+`;
+
+const ScrollAnchor = styled.div`
+  height: 1px;
+  width: 100%;
+`;
+
+const ScrollToBottomButton = styled.button<{ visible: boolean }>`
+  position: absolute;
+  bottom: 24px;
```

Comment:
we should try to avoid using `px`, it is not responsive. should use `rem` instead.

but in this case I think it's working at least on my testing.